### PR TITLE
Map relationship code  case 1366

### PR DESCRIPTION
**Pull request by mlipardo**
This maps out the client relations type code to the system code so we can have a description of what the relationship between from Child-Client to a Client.
[Map The Relationship System codes CASE-1366](https://osi-cwds.atlassian.net/browse/CASE-1366).

### CODE REVIEW

File: lib/placements/address_map_service.rb [hunk 1]
```diff
@@ -12,13 +12,25 @@ def initialize(address_repository = Addresses::AddressRepository.new,
...
       decorate_with_addresses(related_clients, token)
     end
 
     private
 
+    def decorate_with_relationship_type(relationships)
+      relationships.map do |relationship|
+        relationship_attribute = relationship.to_h
+        client = relationship.related_client.to_h
+        @type = relationship_attribute[:relationship_type_code]
```
Comment (edgibbs) [inline lib/placements/address_map_service.rb]: this should just be a variable, not an instance variable

File: lib/relationships/relationship.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@ class Relationship < Dry::Struct
     attribute :relationship_id, Types::String
     attribute :client_id, Types::String.optional
     attribute :related_client_id, Types::String.optional
-    attribute :relationship_type_code, Types::String.optional
+    attribute :relationship_type_code, Types::Int.optional
```
Comment (edgibbs) [inline lib/relationships/relationship.rb]: like the more exact typing here.

Comment (mlipardo) [reply lib/relationships/relationship.rb]: 👍

**Comment by edgibbs** [comment]
So the code climate issues are acceptable for now it's just modeling lookup hashes.  Will be replaced by future story that contains all the lookup codes loaded at startup.

**Pull request merged**